pkg/plotexec/plot_stats.go

```diff
@@ -0,0 +1,97 @@
+package plotexec
+
+import (
+	"github.com/warpfork/warpforge/pkg/workspace"
+	"github.com/warpfork/warpforge/wfapi"
+)
+
+// Might not match the package name -- funcs in this file certainly don't exec anything.
+
+type PlotStats struct {
+	InputsUsingCatalog      int
+	InputsUsingIngest       int
+	InputsUsingMount        int
+	ResolvableCatalogInputs int
+	ResolvedCatalogInputs   map[wfapi.CatalogRef]wfapi.WareID // might as well remember it if we already did all that work.
+	UnresolvedCatalogInputs map[wfapi.CatalogRef]struct{}
+}
+
+// ComputeStats counts up how many times a plot uses various features,
+// and also checks for reference resolvablity.
+//
+// Any errors arising from this process have to do with failure to load
+// catalog info, and causes immediate abort which will result in
+// incomplete counts for all features.
+//
+// Errors:
+//
+// 	- warpforge-error-catalog-invalid -- like it says on the tin.
+// 	- warpforge-error-catalog-parse -- like it says on the tin.
+// 	- warpforge-error-io -- for IO errors while reading catalogs.
+//
+func ComputeStats(plot wfapi.Plot, wsSet workspace.WorkspaceSet) (PlotStats, error) {
```

Comment:
The is one thing in this PR that should *definitely* be rescued: this ComputeStats function is more correct than where it was hoisted from.  The old occurrence isn't recursive, which is... wrong.